### DISCUSSION TITLE
How can I redirect pages that are not found to the home page?

### DISCUSSION THREAD

**Discussion by mattiapv** [nuxt/nuxt#19133] [Questions] [answered]
How can I avoid displaying the 404 error page and instead automatically redirect to the home page?
For example, if a user navigates to a non-existent route such as "/foo", I want to automatically redirect them to the root page "/", without showing the 404 error page.
I have already added a middleware for this purpose, but I don't want to manually add and check every page.
Is there a more efficient way to accomplish this?

```js
export default defineNuxtRouteMiddleware((to) => {
  if (
    to.path != '/' &&
    to.path != '/contacts' &&
    to.path != '/privacyPolicy'
  ) {
    return navigateTo({ path: '/' });
  }
});
```

**Comment by Baroshem** [comment, marked as answer]
Hey,

What about using something like pathMatch from Vue Router? Or a similar regexp that will search for non existing pages and redirect them to homepage?

https://router.vuejs.org/guide/migration/#removed-star-or-catch-all-routes

You could also try to create a global middleware that would react to error thrown that the page was not found and then redirect to homepage instead.

**Reply by mattiapv** [reply]
I'm not sure how to use Vue Router since the routes are automatically generated from the "pages" directory.
I am using a global middleware called "redirect.global.js" located in the "middleware" folder. 
I searched through the documentation, but I was unable to find information on how to handle errors in middleware. However, I was able to resolve the issue by verifying the "to.name" property.
```js
export default defineNuxtRouteMiddleware((to) => {
  if (!to.name) return navigateTo({ path: '/' });
});
```

**Reply by AloisSeckar** [reply]
> I'm not sure how to use Vue Router

Router is availaible to you via `useRouter` composable - https://nuxt.com/docs/api/composables/use-router
